### PR TITLE
add empty sb intermediate check

### PR DESCRIPTION
**Pull request by oleksandr-didyk**
Addresses issues in https://github.com/dotnet/arcade/pull/13067 by hiding the check behind a flag

PR adding the flag into NuGet's project file in installer - https://github.com/dotnet/installer/pull/17100 
(has to be merged before this PR)

### CODE REVIEW

File: src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildArcade.targets [hunk 1]
```diff
@@ -79,6 +79,10 @@
       <IntermediateNupkgArtifactFile Remove="$(CurrentRepoSourceBuildArtifactsPackagesDir)**\*.symbols.nupkg" />
     </ItemGroup>
 
+    <Error
+      Text="No source-build artifacts were found and the resulting intermediate is empty." 
+      Condition="'$(NoSourceBuildArtifactsProduced)' == '' and '@(IntermediateNupkgArtifactFile)' == ''" />
```
Comment (MichaelSimons) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildArcade.targets:84]: IMO this should be abstract and not coupled to NuGet.  There may be other scenarios in the future where we encounter this.  Something like `$(AllowEmptySourceBuiltIntermediates)` != 'True' feels like a more robust approach.

Comment (oleksandr-didyk) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildArcade.targets:84]: Thanks for the suggestion, that feels like a much better name, will change

**Comment by missymessa** [comment]
@oleksandr-didyk are you still working on this PR?

**Comment by oleksandr-didyk** [comment]
> @oleksandr-didyk are you still working on this PR?

Shoot, I thought this was merged a long time ago. My bad and thanks for noticing, I will finish it

**Pull request merged**